src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -355,6 +366,25 @@ public void download(
     try {
       waitForBulkTransfer(downloads, /* cancelRemainingOnInterrupt=*/ true);
     } catch (Exception e) {
+      if (captureCorruptedOutputsDir != null) {
+        if (e instanceof BulkTransferException) {
+          for (Throwable suppressed : e.getSuppressed()) {
+            if (suppressed instanceof OutputDigestMismatchException) {
+              // Capture corrupted outputs
+              try {
+                String outputPath = ((OutputDigestMismatchException) suppressed).getOutputPath();
+                Path localPath = ((OutputDigestMismatchException) suppressed).getLocalPath();
+                Path dst = captureCorruptedOutputsDir.getRelative(outputPath);
```

Comment:
Is it guaranteed (even with broken or malicious backends) that the output path will never contain `..` references or start with a `/`? Should we check that the resulting path `dst` is still below the `captureCorruptedOutputsDir` just to make sure? 🤔

Reply:
We are safe here since `outputPath` is what we obtained from skyframe and send to remote execution backend (something like `bazel-out/darwin-fastbuild/bin/...`).

It is a path relative to the working directory (or input root depends on the flags) when the remote backend executing the action.

Reply:
However, I added the check in case the invariance changed in the future. Thanks for the catch!